scripts/sct_analyze_lesion.py

```diff
@@ -27,53 +27,65 @@
-                      mandatory=False,
-                      example='t2.nii.gz')
-    parser.add_option(name="-f",
-                      type_value="str",
-                      description="Path to folder containing the atlas/template registered to the anatomical image.",
-                      mandatory=False,
-                      example="./label")
-    parser.add_option(name="-ofolder",
-                      type_value="folder_creation",
-                      description="Output folder",
-                      mandatory=False,
-                      example='./')
-    parser.add_option(name="-r",
-                      type_value="multiple_choice",
-                      description="Remove temporary files.",
-                      mandatory=False,
-                      default_value='1',
-                      example=['0', '1'])
-    parser.add_option(name="-v",
-                      type_value='multiple_choice',
-                      description="Verbose: 0 = nothing, 1 = classic, 2 = expended",
-                      mandatory=False,
-                      example=['0', '1', '2'],
-                      default_value='1')
+
+    parser = argparse.ArgumentParser(
+        description='Compute statistics on lesions of the input binary file (1 for lesion, 0 for background). The function assigns an ID value to each lesion (1, 2, 3, etc.) and outputs morphometric measures for each lesion:'
+                    '\n- volume [mm^3]'
+                    '\n- length [mm]: length along the Superior-Inferior axis'
+                    '\n- max_equivalent_diameter [mm]: maximum diameter of the lesion, when '
+                    '\n  approximating the lesion as a circle in the axial cross-sectional plane '
+                    '\n  orthogonal to the spinal cord'
+                    '\n\nIf an image (e.g. T2w or T1w image, texture image) is provided, it computes the mean and standard deviation values of this image within each lesion.'
+                    '\n\nIf a registered template is provided, it computes:'
+                    '\n- the distribution of each lesion depending on each vertebral level and on each'
+                    '\n  region of the template (e.g. GM, WM, WM tracts).'
+                    '\n- the proportion of ROI (e.g. vertebral level, GM, WM) occupied by lesion.'
+                    '\n\nN.B. If the proportion of lesion in each region (e.g. WM and GM) does not sum '
+                    '\nup to 100%, it means that the registered template does not fully cover the '
+                    '\nlesion, in that case you might want to check the registration results.',
+        add_help=None,
+        formatter_class =argparse.RawDescriptionHelpFormatter,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+    mandatoryArguments = parser.add_argument_group("\nMandatory arguments")
+    mandatoryArguments.add_argument("-m",
+                        help="Lesion mask to analyze (e.g. 't2_lesion.nii.gz')",
```

Comment:
@YangHee-Min check your indentation (i'm fixing it for this file, please correct for the other files)